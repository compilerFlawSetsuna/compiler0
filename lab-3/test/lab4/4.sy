int main(){
    a[100]=x;
    if(a>=n) return break;
    a%b;
    a+b;
    a&&b;
    a=1234.23456;
    return djfasldfjals;
}
