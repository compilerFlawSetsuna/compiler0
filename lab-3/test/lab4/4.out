     token    lexeme    lineno       val
       INT       int         1         0
        ID      main         1         1
    LPAREN         (         1         0
    RPAREN         )         1         0
    LBRACE         {         1         0
        ID         a         2         1
     LSQUA         [         2         0
       INT       100         2       100
     RSQUA         ]         2         0
    ASSIGN         =         2         0
        ID         x         2         1
 SEMICOLON         ;         2         0
        IF        if         3         0
    LPAREN         (         3         0
        ID         a         3         1
 GREATEREQ        >=         3         0
        ID         n         3         1
    RPAREN         )         3         0
    RETURN    return         3         0
     BREAK     break         3         0
 SEMICOLON         ;         3         0
        ID         a         4         1
       MOD         %         4         0
        ID         b         4         1
 SEMICOLON         ;         4         0
        ID         a         5         1
       ADD         +         5         0
        ID         b         5         1
 SEMICOLON         ;         5         0
        ID         a         6         1
       AND        &&         6         0
        ID         b         6         1
 SEMICOLON         ;         6         0
        ID         a         7         1
    ASSIGN         =         7         0
     FLOAT1234.23456         71234.234619
 SEMICOLON         ;         7         0
    RETURN    return         8         0
        IDdjfasldfjals         8         1
 SEMICOLON         ;         8         0
    RBRACE         }         9         0
